parser.go

```diff
@@ -828,12 +838,29 @@ func (parser *Parser) matchTags(comments []*ast.Comment) (match bool) {
 	return true
 }
 
+func (parser *Parser) matchExtension(comments []*ast.Comment) (match bool) {
+	if len(parser.parseExtension) != 0 {
+		for _, comment := range comments {
+			commentLine := strings.TrimSpace(strings.TrimLeft(comment.Text, "/"))
+			fields := FieldsByAnySpace(commentLine, 2)
+			lowerAttribute := strings.ToLower(fields[0])
+
+			if lowerAttribute == fmt.Sprintf("@x-%s", strings.ToLower(parser.parseExtension)) {
+				return true
+			}
+		}
+		return false
+	}
+	return true
+}
+
```

Comment:
The only thing you are using from the parser is parser.ParseExtension, This function is a mimic of in_array `func in_array(val string, array []string) bool` with default to true if the search value is empty. 

Such a simple function should not be attached to the parser.
```go
func matchExtension(textToFind string, comments []*ast.Comment) bool
```

Reply:
Got it.
I was basically c/p above matchTags without too much thinking. you have a good point.

I can detach it from the parser, before that just to double check do you recommend to create a "helper" in_array-type method or still keep it specific to extension parsing? (Not sure if we have such a helper file/methods today somewhere - can't find them in the repo)

Reply:
Just detach the function from the parser.

Reply:
Thanks.